File: src/main/java/org/acg/user_services/repository/PasswordResetTokenRepo.java
package org.acg.user_services.repository;

import org.acg.user_services.repository.entity.PasswordResetToken;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.sql.Date;

@Repository
public interface PasswordResetTokenRepo extends CrudRepository<PasswordResetToken, Long> {
    PasswordResetToken findByToken(String token);
    void deleteAllByExpireDateBefore(Date date);
}

File: src/main/kotlin/org/acg/user/repository/PasswordResetTokenRepo.kt
package org.acg.user.repository;

import org.acg.user.repository.entity.PasswordResetToken;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.sql.Date;

@Repository
interface PasswordResetTokenRepo : CrudRepository<PasswordResetToken, Long> {
    fun findByToken(token: String) : PasswordResetToken?
    fun deleteAllByExpireDateBefore(date: Date)
}
